exchange/auction.go

```diff
@@ -176,6 +175,52 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	return errs
 }
 
+func (a *auction) cacheBid(bidToCache *pbsOrtbBid, incBannerBids bool, incVastBids bool, hasCustomCacheKey bool, toCache *[]prebid_cache_client.Cacheable, expByImp map[string]int64, defaultTTLs *config.DefaultTTLs, ttlBuffer int64, bidCategory map[string]string, hbCacheID string) (bool, bool, int, error) {
+	var chachedBid, chachedVast bool = false, false
+	var cachedSoFar int = len(*toCache)
+	var anError error = nil
+
+	if incBannerBids { //banner
+		if jsonBytes, err := json.Marshal(bidToCache.bid); err == nil {
+			if hasCustomCacheKey {
+				anError = errors.New("cannot use custom cache key for non-vast bids")
+			}
+			*toCache = append(*toCache, prebid_cache_client.Cacheable{
+				Type:       prebid_cache_client.TypeJSON,
+				Data:       jsonBytes,
+				TTLSeconds: cacheTTL(expByImp[bidToCache.bid.ImpID], bidToCache.bid.Exp, defTTL(bidToCache.bidType, defaultTTLs), ttlBuffer),
+			})
+			chachedBid = true
+		} else {
+			anError = err
+		}
+
+	}
+	if incVastBids && bidToCache.bidType == openrtb_ext.BidTypeVideo { //video
+		if jsonBytes, err := json.Marshal(makeVAST(bidToCache.bid)); err == nil {
+			_, isTopBid := a.impsToTopBids[bidToCache.bid.ImpID]
+			if catDur, ok := bidCategory[a.impsToTopBids[bidToCache.bid.ImpID].bid.ID]; ok && isTopBid {
+				*toCache = append(*toCache, prebid_cache_client.Cacheable{
+					Type:       prebid_cache_client.TypeXML,
+					Data:       jsonBytes,
+					TTLSeconds: cacheTTL(expByImp[bidToCache.bid.ImpID], bidToCache.bid.Exp, defTTL(bidToCache.bidType, defaultTTLs), ttlBuffer),
+					Key:        fmt.Sprintf("%s_%s", catDur, hbCacheID),
```

Comment:
It is possible for hbCacheID to not be set, in which case a problem occurs here.  This possibility needs to be avoided.

Reply:
No it isn't, the current logic prevents it